torch/csrc/jit/fusion_compiler.cpp

```
@@ -122,27 +125,22 @@ static void emitIndexingFor(std::ostream & out, const std::string & tensor, int
   }
 }
 
-static std::ostream& operator<<(std::ostream & out, const TensorDesc & d) {
-  out << d.scalar_type << "[";
-  for(auto b : d.contiguity)
-    out << b << ";";
-  out << "]";
-  return out;
-}
-
-static std::string nodeName(Node * n) {
-  return "n"+std::to_string(n->unique());
+std::string nodeName(Node * n) {
+  return "n" + std::to_string(n->unique());
 }
 
 // TODO: we need to support double-precision
-static std::unordered_map<NodeKind,std::string> simple_map_ops = {
-  {kSigmoid,         "1.f / (1.f + expf(-${0}))"},
-  {kTanh,            "tanhf(${0})"},
-  {kMul,             "${0} * ${1}"},
-  {kAdd,             "${0} + ${1}"},
+std::unordered_map<NodeKind,std::function<std::string(Node*)>> simple_map_ops = {
+  {kSigmoid,         [](Node*) { return "1.f / (1.f + expf(-${0}))"; }},
+  {kTanh,            [](Node*) { return "tanhf(${0})"; }},
+  {kMul,             [](Node*) { return "${0} * ${1}"; }},
+  {kAdd,             [](Node*) { return "${0} + ${1}"; }},
+  {kNeg,             [](Node*) { return "(-${0})"; }},
+  // TODO: support both float and int constants
+  {kAddConstant,     [](Node* n) { return std::to_string(n->f(kvalue)) + " + ${0}"; }},
```

If I trip on this, will it silently return 0.0 or raise an error?

---

It will raise an error. We never create AddConstant nodes with int values at the moment so it's not an issue, but we might want to start at some point.